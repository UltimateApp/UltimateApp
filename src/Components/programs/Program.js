import React, { useState } from 'react';
import { StyleSheet, TouchableOpacity, View, Text, FlatList } from 'react-native';

import theme from '../../styles/theme.style';

const Program = props => {
  const { id, title, trainings } = props.program;
  const firstTodoTrainingIndex = 0; // Faking it for now

  const onPress = item =>
    props.navigation.navigate('TrainingPage', { training: trainings[firstTodoTrainingIndex], program: props.program });
  const width = `${(firstTodoTrainingIndex * 100) / trainings.length}%`;

  return (
    <TouchableOpacity style={styles.program} onPress={() => onPress()}>
      <Text style={styles.programTitle}>{title}</Text>
      <Text style={styles.completion}>0/{trainings.length}</Text>
      <View style={styles.progressBar}>
        <View style={[StyleSheet.absoluteFill, styles.fillProgressBar, { width }]} />
      </View>
    </TouchableOpacity>
  );
};

export default Program;

const styles = StyleSheet.create({
  program: {
    marginBottom: 10,
    padding: 20,
    backgroundColor: theme.COLOR_SECONDARY_LIGHT,
  },
  programTitle: {
    fontSize: theme.FONT_SIZE_LARGE,
    fontWeight: 'bold',
  },
  completion: {
    textAlign: 'right',
    fontWeight: 'bold',
  },
  progressBar: {
    marginTop: 5,
    height: 10,
    width: '100%',
    backgroundColor: 'white',
    borderRadius: 5,
  },
  fillProgressBar: {
    width: '50%',
    borderRadius: 5,
    backgroundColor: theme.COLOR_PRIMARY,
  },
});
