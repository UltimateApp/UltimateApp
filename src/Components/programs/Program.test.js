import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';

import { createProgram } from '../../Fixtures/TestFixtures';
import Program from './Program';

afterEach(cleanup);

describe('<Program />', () => {
  it('renders correctly', () => {
    const program = createProgram();
    const tree = renderer.create(<Program program={program} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
