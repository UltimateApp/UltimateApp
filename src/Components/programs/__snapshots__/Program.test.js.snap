// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Program /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "marginBottom": 10,
      "padding": 10,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "opacity": 1,
        "paddingLeft": 40,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexBasis": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Beginner
      </Text>
      <Text
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        0
        /
        1
         trainings
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#f1f1f1",
            "borderRadius": 2,
            "height": 3,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "#232b2b",
                "borderRadius": 5,
                "width": "50%",
              },
              Object {
                "width": "0%",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": "10%",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "aspectRatio": 0.4560669456066946,
            "height": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Program /> renders correctly when finished 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "marginBottom": 10,
      "padding": 10,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "opacity": 1,
        "paddingLeft": 40,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexBasis": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Beginner
      </Text>
      <Text
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        2
        /
        2
         trainings
         👍🎉
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#f1f1f1",
            "borderRadius": 2,
            "height": 3,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "#232b2b",
                "borderRadius": 5,
                "width": "50%",
              },
              Object {
                "width": "100%",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": "10%",
          "paddingLeft": 20,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "aspectRatio": 0.4560669456066946,
            "height": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
